### PR TITLE
Add organization parameter to GetAuthorizationURL

### PR DESCRIPTION
**Pull request by sheldonvaughn**
Add organization parameter to GetAuthorizationURL function and include a notice message in the function for users who are strictly calling via domain and not any other parameter to identify the connection.

### CODE REVIEW

File: lib/SSO.php [hunk 1]
```diff
@@ -28,6 +29,7 @@ public function getAuthorizationUrl(
         $state,
         $provider,
         $connection,
+        $organization,
```
Comment (robframpton) [inline lib/SSO.php:32]: I think in order to make this a non-breaking change we'll need to default this to null.

```suggestion
        $organization = null,
```
```suggestion
        $organization,
```

File: tests/WorkOS/SSOTest.php [hunk 2]
```diff
@@ -25,6 +25,7 @@ public function testAuthorizationURLExpectedParams(
         $state,
         $provider,
         $connection,
+        $organization,
```
Comment (robframpton) [inline tests/WorkOS/SSOTest.php:28]:
```suggestion
        $organization = null,
```

Comment (robframpton) [reply tests/WorkOS/SSOTest.php:28]: ^ should fix the tests in CI

File: lib/SSO.php [hunk 3]
```diff
@@ -28,17 +29,24 @@ public function getAuthorizationUrl(
...
+        if (!isset($domain) && !isset($provider) && !isset($connection) && !isset($organization)) {
+            $msg = "Either \$domain, \$provider, \$connection, or \$organization is required";
 
             throw new Exception\UnexpectedValueException($msg);
         }
 
+        if (isset($domain) && !isset($provider) && !isset($connection) && !isset($organization)) {
+            $msg = "Domain is being deprecated, please switch to using Connection or Organization ID";
+
+            trigger_error($msg, E_USER_NOTICE);
```
Comment (robframpton) [inline lib/SSO.php:47]: Does this throw a true exception or is it just a warning? We'll want it to be just a warning so maintain backwards compatibility.

Comment (sheldonvaughn) [reply lib/SSO.php:47]: @robframpton  From my research there should be 3 tiers of exception:

E_USER_NOTICE
E_USER_WARNING
E_USER_ERROR

Notice and warning are just messages that won't stop the functionality - https://beamtic.com/e_user_notice-php

Comment (sheldonvaughn) [reply lib/SSO.php:47]: Here's the official PHP documentation https://www.php.net/manual/en/errorfunc.constants.php

Comment (robframpton) [reply lib/SSO.php:47]: Awesome thanks for the docs 👍🏻

**Review by robframpton — APPROVED** [review]

**Review by robframpton — APPROVED** [review]

**Pull request merged**